Prototyping/Sit/sit.js

```diff
@@ -0,0 +1,692 @@
+                    }
+
+                    var timePassed = now - overlayPreSitStart;
+                    if (timePassed >= OVERLAY_PRESIT_FRAME_DURATION) {
+                        overlayPreSitStart = now;
+                        index = index + 1;
+                        updateObj.url = overlayPreSitLoaded[index].url;
+                    }
+
+                    Overlays.editOverlay(overlayPreSit, updateObj);
+                    Overlays.editOverlay(overlayPreSitText, {
+                        rotation: cameraRotation,
+                        position: Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: -0.05, z: -1 })),
+                    });
+                }
+            },
+
+        },
+
+        standUp: {
+            create: function () {
+                if (overlayStandUp === null) {
+                    overlayStandUp = Overlays.addOverlay("image3d", {
+                        position: Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: 0, z: -1 })),
+                        rotation: Camera.orientation,
+                        parentId: MyAvatar.sessionUUID,
+                        parentJoint: "Head",
+                        dimensions: { x: 0.2, y: 0.2 },
+                        url: OVERLAY_URL_STANDUP,
+                        ignoreRayIntersection: false,
+                        drawInFront: true,
+                        visible: true,
+                        emissive: true
+                    });
+                }
+            },
+            remove: function () {
+                if (overlayStandUp) {
+                    Overlays.deleteOverlay(overlayStandUp);
+                    overlayStandUp = null;
+                }
+            },
+            update: function () {
+                if (overlayStandUp) {
+                    Overlays.editOverlay(overlayStandUp, { // too inefficient
+                        rotation: Camera.orientation,
+                        position: Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: 0, z: -1 }))
+                    });
+                }
+            }
+        },
+
+        showSittable: function () { // Show overlays when I'm close to the seat
+
+            if (isInEditMode()) {
+                return;
+            }
+
+            if (overlaySittable === null) { // Make an overlay if there isn't one
+                if (this.sittable.shouldShow()) {
+                    this.sittable.create();
+                    if (overlayIntervalTransparency === null) {
+                        this.sittable.lerpTransparency();
+                    }
+                }
+            } else if (!utils.canSitDesktop()) {
+                this.sittable.remove();
+            }
+        }
+    };
+
+    var utils = {
+
+        // Is the seat used
+        checkSeatForAvatar: function () {
+            return deadlockGate;
```

Comment:
I'm confused is this used to check if the MyAvatar is in the chair? or if ANY avatar is sitting in the chair.